relentless/simulate/lammps.py

```diff
@@ -0,0 +1,290 @@
+        cmds = ['units {style}'.format(style=self.units),
+                'boundary p p p',
+                'atom_style {style}'.format(style=self.atom_style),
+                'read_data {filename}'.format(filename=filename), #TODO: keyword args?
+                self.attach_potentials(sim)]
+
+        return cmds
+
+class InitializeRandomly(Initialize):
+    def __init__(self, neighbor_buffer, seed, units='lj', atom_style='atomic'):
+        super().__init__(neighbor_buffer)
+        self.seed = seed
+        self.units = units
+        self.atom_style = atom_style
+
+    def to_commands(self, sim):
+        cmds = ['units {style}'.format(style=self.units),
+                'boundary p p p',
+                'atom_style {style}'.format(style=self.atom_style)]
+
+        # make box from ensemble
+        box = self.extract_box_params(sim)
+        if not np.all(np.isclose(box[-3:],0)):
+            cmds += ['region box prism {} {} {} {} {} {} {} {} {}'.format(*box)]
+        else:
+            cmds += ['region box block {} {} {} {} {} {}'.format(*box[:-3])]
+        cmds += ['create_box {N} box'.format(N=len(sim.ensemble.types))]
+
+        # use lammps random initialization routines
+        for i in sim.ensemble.types:
+            cmds += ['create_atoms {typeid} random {N} {seed} box'.format(typeid=sim.type_map[i],
+                                                                          N=sim.ensemble.N[i],
+                                                                          seed=self.seed+sim.type_map[i]-1)]
+        cmds += ['mass * 1.0',
+                 'velocity all create {temp} {seed}'.format(temp=sim.ensemble.T,
+                                                            seed=self.seed)]
+
+        cmds += self.attach_potentials(sim)
+
+        return cmds
+
+class MinimizeEnergy(simulate.SimulationOperation):
+    def __init__(self, energy_tolerance, force_tolerance, max_iterations):
+        self.energy_tolerance = energy_tolerance
+        self.force_tolerance = force_tolerance
+        self.max_iterations = max_iterations
+
+    def to_commands(self, sim): #TODO: min_style, min_modfy?
+        cmds += ['minimize {etol} {ftol} {maxiter}'.format(etol=self.energy_tolerance,
+                                                           ftol=self.force_tolerance,
+                                                           maxiter=self.max_iterations)]
+
+        return cmds
+
+# Brownian dynamics not supported by LAMMPS
+
+class AddLangevinIntegrator(simulate.SimulationOperation):
+    def __init__(self, idx, group_idx, t_start, t_stop, damp, seed): #TODO: condense keywords, reconcile with generic?
+        self.idx = idx
+        self.group_idx = group_idx
+        self.t_start = t_start
+        self.t_stop = t_stop
+        self.damp = damp
+        self.seed = seed
+
+    def to_commands(self, sim):
+        cmds += ['fix {idx} {group_idx} langevin {t_start} {t_stop} {damp} {seed}'.format(idx=self.idx,
```

Comment:
This needs another fix command that creates an NVE integrator for `all`, as on its own fix langevin is not an integrator. Also, if we want to support different friction coefficients for each type (as in hoomd), we would need to make a separate `fix langevin` for each type, and make it work on each type group individually.